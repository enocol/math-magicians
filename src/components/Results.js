import React from 'react';
import PropTypes from 'prop-types';

const Results = (props) => {
  const { value } = props;
  return <div className="results gray-bg">{value}</div>;
};

export default Results;

Results.propTypes = {
  value: PropTypes.number.isRequired,
};
