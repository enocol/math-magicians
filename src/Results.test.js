import React from 'react';
import { render } from '@testing-library/react';
import Results from './components/Results';

describe('Results component', () => {
  it('renders correctly with a number value', () => {
    const { asFragment } = render(<Results value={42} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with a different number value', () => {
    const { asFragment } = render(<Results value={123} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
