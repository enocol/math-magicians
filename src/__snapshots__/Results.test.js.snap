// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component renders correctly with a different number value 1`] = `
<DocumentFragment>
  <div
    class="results gray-bg"
  >
    123
  </div>
</DocumentFragment>
`;

exports[`Results component renders correctly with a number value 1`] = `
<DocumentFragment>
  <div
    class="results gray-bg"
  >
    42
  </div>
</DocumentFragment>
`;
